src/Menu/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { borderRadius, color, space } from '@core-ds/primitives';
+import * as React from 'react';
+import styled from 'styled-components';
+import {
+   MenuButtonProps,
+   MenuContext,
+   MenuItemProps,
+   MenuListProps,
+   MenuProps,
+   useMenu,
+   useMenuButton,
+   useMenuItem,
+   useMenuList,
+} from './useMenu';
+
+export function Menu(props: MenuProps) {
+   const { children, className } = props;
+   const menu = useMenu(props);
+
+   return (
+      <MenuContext.Provider value={menu}>
+         <MenuContainer className={className}>{children}</MenuContainer>
+      </MenuContext.Provider>
+   );
+}
+
+const MenuContainer = styled.div`
+   position: relative;
+`;
+
+function UnstyledMenuButton(props: MenuButtonProps) {
+   const menuButtonProps = useMenuButton(props);
+   return <button {...menuButtonProps} />;
+}
+
+export const MenuButton = styled(UnstyledMenuButton)`
+   display: flex;
+   align-items: center;
+   background: none;
+   border: none;
+   color: ${color.white};
+   font-size: 14px;
+   cursor: pointer;
+   outline: none;
+   border-radius: ${borderRadius.md};
+   font-family: inherit;
+   padding: 0;
+   &:focus {
+      box-shadow: 0 0 0 0.2rem rgb(25 117 241 / 25%);
+   }
+`;
+
+export function UnstyledMenuList(props: MenuListProps) {
+   const menuListProps = useMenuList(props);
+
+   return <ul {...menuListProps} />;
+}
+
+export const MenuList = styled(UnstyledMenuList)`
+   margin: 0;
+   margin-bottom: ${space[1]};
+   padding: ${space[2]} 0;
+   border-radius: ${borderRadius.lg};
+   list-style: none;
+   background-color: white;
+   color: black;
+   display: none;
+   position: absolute;
+   transform: translateY(-100%) translateY(-${space[2]});
+   top: 0;
+   left: 0;
+   min-width: 224px;
+   &[data-active='true'] {
+      display: block;
+   }
+`;
+
+function UnstyledMenuItem(props: MenuItemProps) {
+   const itemProps = useMenuItem(props);
+   if (props.url != null) {
```

Comment:
Yes. Basically if you want to check for `null` or `undefined` you'll want use `!= null` (or `== null`) as it is the most precise operator (e.g. see lodash implementation of [isNil](https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L11973)).

I think you mean `props.url` not `!props.url`. `props.url` in this specific case works equally well. `props !== null` won't work because it only check for `null` but not for `undefined`.

The reason I default to use `== null` is that it avoids subtle bugs. For example when the value you are checking is a numeric prop and you want to check for its presence (hence it is not null or undefined), using `!value` won't work because it's true also when value equal to 0. So it frees your mind from having to think each time if `if (value)` falls for an edge case.

Is there any particular reason why you prefer `props.url` ?

Reply:
It's just shorter but your reasons are totally valid.

Reply:
I guess also coming from legacy js code, I always give a suspicious look at != or == null, since I usually expect to see === or !==. But I think checking for null or undefined with == null makes perfect sense so maybe I'll start doing that.